### ISSUE TITLE
No such label 'IL_??' within the scope of the goto statement in switch

### ISSUE THREAD

**Issue by Albeoris** [0xd4d/dnSpy#352]
If there is no code after a switch statement we can get a such error.

**Sample:**
```
0	0000	call	valuetype [UnityEngine]UnityEngine.SystemLanguage [UnityEngine]UnityEngine.Application::get_systemLanguage()
1	0005	stloc.0
2	0006	ldloc.0
3	0007	ldc.i4.s	10
4	0009	sub
5	000A	switch	[16 (0044), 6 (0027), 6 (0027), 6 (0027), 22 (0056), 20 (0050)]
6	0027	ldloc.0
7	0028	ldc.i4.s	0x15
8	002A	beq	24 (005C) ldstr "it"
9	002F		ldloc.0
10	0030	ldc.i4.s	0x16
11	0032	beq	18 (004A) ldstr "jp"
12	0037	ldloc.0
13	0038	ldc.i4.s	0x22
14	003A	beq	26 (0062) ldstr "es"
15	003F	br	28 (0068) ldstr "us"
16	0044	ldstr	"us"
17	0049	ret
18	004A	ldstr	"jp"
19	004F		ret
20	0050	ldstr	"gr"
21	0055	ret
22	0056	ldstr	"fr"
23	005B	ret
24	005C	ldstr	"it"
25	0061	ret
26	0062	ldstr	"es"
27	0067	ret
28	0068	ldstr	"us"
29	006D	ret
```

```
private string GetSystemLanguageSymbol()
	{
		SystemLanguage systemLanguage = Application.systemLanguage;
		switch (systemLanguage)
		{
		case SystemLanguage.English:
			return "us";
		case SystemLanguage.Estonian:
		case SystemLanguage.Faroese:
		case SystemLanguage.Finnish:
			IL_27:
			if (systemLanguage == SystemLanguage.Italian)
			{
				return "it";
			}
			if (systemLanguage == SystemLanguage.Japanese)
			{
				return "jp";
			}
			if (systemLanguage != SystemLanguage.Spanish)
			{
				return "us";
			}
			return "es";
		case SystemLanguage.French:
			return "fr";
		case SystemLanguage.German:
			return "gr";
		}
		goto IL_27; // Cannot resolve symbol 'IL_27'
	}
```

Possible fix: Remove an conditional in the AndroidExpansionUI.FindConditions.
```
// Heuristis to determine if we want to use fallthough as default case
if (fallTarget != null && !frontiers.Contains(fallTarget)) {
	HashSet<ControlFlowNode> content = FindDominatedNodes(scope, fallTarget);
	/*if (content.Any())*/ { // Always add a defualt block
		var caseBlock = new ILSwitch.CaseBlock() { EntryGoto = new ILExpression(ILCode.Br, fallLabel) };
		ilSwitch.CaseBlocks.Add(caseBlock);
		tail = block.Body.RemoveTail(ILCode.Br);
		if (context.CalculateBinSpans)
			tail[0].AddSelfAndChildrenRecursiveBinSpans(caseBlock.BinSpans);
		
		scope.ExceptWith(content);
		caseBlock.Body.AddRange(FindConditions(content, fallTarget));
		// Add explicit break which should not be used by default, but the goto removal might decide to use it
		caseBlock.Body.Add(new ILBasicBlock() {
			Body = {
				new ILLabel() { Name = "SwitchBreak_" + (nextLabelIndex++).ToString() },
				new ILExpression(ILCode.LoopOrSwitchBreak, null)
			}
		});
	}
}
```

I think it's a bad idea, but I cannot offer nothing better.

![default](https://cloud.githubusercontent.com/assets/4787832/22406710/eb0c99a0-e668-11e6-8a59-36ec91739c99.png)

**Comment by SlowLogicBoy** [comment]
This is decompiler bug. Try latest build, it might be fixed there, if not, we will need to wait for ILSpy's new decompiler.

**Comment by 0xd4d** [comment]
Use latest build, it fixes this thing and C#7 state machines.
